Bottom: 75bbfd3f6334c2985085f9f5bafea533ccb4d6b4
Top:    666abc91626127b8682a7c4c563fb0555ea9c6e7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-03 16:53:45 +0100

Refresh of work-in-progress-on-subcomposi

---

diff --git a/src/draw.c b/src/draw.c
index 0ab3ece..991fd8f 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1153,8 +1153,7 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
   int n_entries = PCB->LayerGroups.Number[group];
   Cardinal *layers = PCB->LayerGroups.Entries[group];
 
-  for (i = n_entries - 1; i >= 0;
-      i--, gui->set_layer (0, group, 0)) /* HACK: Subcomposite each layer in a layer group separately */
+  for (i = n_entries - 1; i >= 0; i--)
     {
       layernum = layers[i];
       Layer = PCB->Data->Layer + layers[i];
@@ -1163,6 +1162,9 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
         rv = 0;
       if (layernum < max_copper_layer && Layer->On)
         DrawLayerCommon (Layer, drawn_area, true);
+
+      if (gui->gui) /* HACK: Subcomposite each layer in a layer group separately */
+        gui->set_layer (0, group, 0);
     }
   if (n_entries > 1)
     rv = 1;
